staging/src/k8s.io/client-go/features/features.go

```diff
@@ -110,13 +110,13 @@ func ReplaceFeatureGates(newFeatureGates Gates) {
 func replaceFeatureGatesWithWarningIndicator(newFeatureGates Gates) bool {
 	shouldProduceWarning := false
 
-	if defaultFeatureGates, ok := FeatureGates().(*envVarFeatureGates); ok {
+	if defaultFeatureGates, ok := FeatureGates().(*reloadableFeatureGates).delegate.Load().(*featureGatesWrapper).Gates.(*envVarFeatureGates); ok {
```

Comment:
we could create a method that could validate if type assertion is possible each time we need to cast.
not being able to cast to `*reloadableFeatureGates` and `*featureGatesWrapper` should panic